proto/asgt/v2/dataset_service.proto

```diff
@@ -126,7 +132,15 @@ message TruncateDatasetRequest {
   string name = 1 [(validate.rules).string = { pattern: "^[A-Za-z0-9.][A-Za-z0-9_.>-]*$", max_bytes: 256 }]; // text-no-spaces
 }
 
-message TrainingsRequest {
+message GetDatasetTrainingsRequest {
   string dataset_name = 1 [(validate.rules).string = { pattern: "^[A-Za-z0-9.][A-Za-z0-9_.>-]*$", max_bytes: 256 }]; // text-no-spaces
-  uint32 number_of_trainings = 2;
+  int64 limit = 2; // mapped to URL query parameter `limit`
```

Comment:
I think I'd like this more to be an "option" message containing the limit... e.g. 

```
message GetDatasetTrainingsRequest {
  message TrainingRequestOptions {
    int64 limit = 1;
  }
  string dataset_name = 1 [val];
  message TrainingRequestOptions options = 2;
```

Reply:
No specific reason behind it.. just saw it being done that way

Reply:
Aaah I get now what you meant with the `options`. Yes, it would particularly make sense if we would have more calls that need the same options - so maybe not now, but in the future it could be useful. I see if I can do it before merging this. I'm just not sure how the URL query parameters will like the nested messages... but it should be possible.

Reply:
well "maybe in the future" isn't backwards compatible 😅 

Also IDK how all of this works with query params as this is completely new to me